ocs_ci/ocs/resources/mcg.py

```diff
@@ -799,20 +799,41 @@ def _check_mirroring():
                     else:
                         results.append(False)
 
-            return all(results)
-
-        try:
-            for mirroring_is_complete in TimeoutSampler(timeout, 5, _check_mirroring):
-                if mirroring_is_complete:
+            return results
+
+        is_mirroring_complete = _check_mirroring()
+        current_percentage = (
+            is_mirroring_complete.count(True) / len(is_mirroring_complete)
+        ) * 100
+        logger.info(f"{current_percentage}% mirroring is done.")
+        previous_percentage = 0
+        if current_percentage == 100:
+            logger.info("All objects mirrored successfully.")
+            return True
+        else:
```

Comment:
If current_percentage == 100 at line 810, we're exiting the method with the return True statement so there's no need to use an else: block that introduces an unnecessary additional level of indentation.

Also, the condition of the `while` statement at line 814 saves you the need for this if check entirely - exiting the while loop indicates that the mirroring has been complete, so just log the result and return True after it. This also applies for the `if` check at lines 834-836, so this helps to apply the DRY principle as well (let's say we want to change the logging info to include some data in the future here - you would have to change it in two places under this current implementation).

Reply:
Fixed